lib/sql.js

```diff
@@ -877,13 +945,38 @@ SQLConnector.prototype.buildOrderBy = function(model, order) {
  * @returns {{names: Array, values: Array, properties: Array}}
  */
 SQLConnector.prototype.buildFields = function(model, data, excludeIds) {
+//  var props = this.getModelDefinition(model).properties;
+  var keys = Object.keys(data);
+  return this._createFields(model, data, keys, excludeIds);
+};
+
+/**
+ * Build an array of fields for the replace database operation
+ * @param {String} model Model name
+ * @param {Object} data Model data object
+ * @param {Boolean} excludeIds Exclude id properties or not, default to false
+ * @returns {{names: Array, values: Array, properties: Array}}
+ */
+SQLConnector.prototype._buildReplaceFields = function(model, data, excludeIds) {
+  var props = this.getModelDefinition(model).properties;
+  var keys = Object.keys(props);
+  return this._createFields(model, data, keys, excludeIds);
+};
+
+/*
+ * @param {String} model The model name.
+ * @returns {Object} data The model data object.
+ * @returns {Array} keys The key fields for which need to be built.
+ * @param {Boolean} excludeIds Exclude id properties or not, default to false
+ * @private
+ */
+SQLConnector.prototype._createFields = function(model, data, keys, excludeIds) {
```

Comment:
I find `buildFields` vs. `createFields` a bit confusing, the distinction is not clear to me. Could you please use a more descriptive name, e.g. `_buildFieldsForKeys`?